regression importing cygrpc grpc python 1.24.0

### What version of gRPC and what language are you using?
1.24.0 python

### What operating system (Linux, Windows,...) and version?
Windows

### What runtime / compiler are you using (e.g. python version or version of gcc)
Python 3.6.5 from https://www.python.org/ftp/python/3.6.5/python-3.6.5.exe

### What did you do?
If possible, provide a recipe for reproducing the error. Try being specific and include code snippets if helpful.
Imported the google-cloud-monitoring 0.33.0 library.

### What did you expect to see?
My test runs.

### What did you see instead?
```python
my_test.py:7: in <module>
    from google.cloud import monitoring
Python36\lib\site-packages\google\cloud\monitoring.py:20: in <module>
    from google.cloud.monitoring_v3 import AlertPolicyServiceClient
Python36\lib\site-packages\google\cloud\monitoring_v3\__init__.py:21: in <module>
    from google.cloud.monitoring_v3.gapic import alert_policy_service_client
Python36\lib\site-packages\google\cloud\monitoring_v3\gapic\alert_policy_service_client.py:26: in <module>
    import google.api_core.gapic_v1.config
Python36\lib\site-packages\google\api_core\gapic_v1\__init__.py:16: in <module>
    from google.api_core.gapic_v1 import config
Python36\lib\site-packages\google\api_core\gapic_v1\config.py:23: in <module>
    import grpc
Python36\lib\site-packages\grpc\__init__.py:23: in <module>
    from grpc._cython import cygrpc as _cygrpc
E   ImportError: DLL load failed: The specified module could not be found.
```

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

See [TROUBLESHOOTING.md](https://github.com/grpc/grpc/blob/master/TROUBLESHOOTING.md) for how to diagnose problems better.

### Anything else we should know about your project / environment?
This was working with grpcio-1.23.0. I successfully worked around the issue with 1.24.0 by upgrading to Python 3.7.4 from https://www.python.org/ftp/python/3.7.4/python-3.7.4.exe

--- a/setup.py
+++ b/setup.py
@@ -147,16 +147,21 @@ EXTRA_ENV_COMPILE_ARGS = os.environ.get('GRPC_PYTHON_CFLAGS', None)
 EXTRA_ENV_LINK_ARGS = os.environ.get('GRPC_PYTHON_LDFLAGS', None)
 if EXTRA_ENV_COMPILE_ARGS is None:
   EXTRA_ENV_COMPILE_ARGS = ' -std=c++11'
-  if 'win32' in sys.platform and sys.version_info < (3, 5):
-    EXTRA_ENV_COMPILE_ARGS += ' -D_hypot=hypot'
-    # We use define flags here and don't directly add to DEFINE_MACROS below to
-    # ensure that the expert user/builder has a way of turning it off (via the
-    # envvars) without adding yet more GRPC-specific envvars.
-    # See https://sourceforge.net/p/mingw-w64/bugs/363/
-    if '32' in platform.architecture()[0]:
-      EXTRA_ENV_COMPILE_ARGS += ' -D_ftime=_ftime32 -D_timeb=__timeb32 -D_ftime_s=_ftime32_s'
+  if 'win32' in sys.platform:
+    if sys.version_info < (3, 5):
+      EXTRA_ENV_COMPILE_ARGS += ' -D_hypot=hypot'
+      # We use define flags here and don't directly add to DEFINE_MACROS below to
+      # ensure that the expert user/builder has a way of turning it off (via the
+      # envvars) without adding yet more GRPC-specific envvars.
+      # See https://sourceforge.net/p/mingw-w64/bugs/363/
+      if '32' in platform.architecture()[0]:
+        EXTRA_ENV_COMPILE_ARGS += ' -D_ftime=_ftime32 -D_timeb=__timeb32 -D_ftime_s=_ftime32_s'
+      else:
+        EXTRA_ENV_COMPILE_ARGS += ' -D_ftime=_ftime64 -D_timeb=__timeb64'
     else:
-      EXTRA_ENV_COMPILE_ARGS += ' -D_ftime=_ftime64 -D_timeb=__timeb64'
+      # We need to statically link the C++ Runtime, only the C runtime is
+      # available dynamically
+      EXTRA_ENV_COMPILE_ARGS += ' /MT'
   elif "linux" in sys.platform:
     EXTRA_ENV_COMPILE_ARGS += ' -std=gnu99 -fvisibility=hidden -fno-wrapv -fno-exceptions'
   elif "darwin" in sys.platform:

